.github/workflows/build-test-publish.yml

```diff
@@ -0,0 +1,190 @@
+      - name: Install MSys2 and dependencies
+        uses: msys2/setup-msys2@v2
+        with:
+          update: true
+          msystem: MINGW64
+          install: >-
+            mingw-w64-x86_64-curl
+            unzip
+            mingw-w64-x86_64-python
+            mingw-w64-x86_64-python-pip
+            mingw-w64-x86_64-gtksourceview3
+            mingw-w64-x86_64-python-pillow
+            mingw-w64-x86_64-enchant
+            mingw-w64-x86_64-hunspell-en
+            mingw-w64-x86_64-python-cairo
+            mingw-w64-x86_64-gtk3
+            mingw-w64-x86_64-python3-gobject
+            mingw-w64-x86_64-python-cffi
+            mingw-w64-x86_64-hunspell
+            mingw-w64-x86_64-hunspell-en
+            mingw-w64-x86_64-openssl
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Rewrite version for dev if not tag
+        if: "!startsWith(github.ref, 'refs/tags/')"
+        shell: msys2 {0}
+        run: |
+          perl -i -pe "s/__version__\s*=\s*'(.*?)(rc.*|a.*|post.*)?'/__version__='\1.dev${GITHUB_SHA::8}'/" setup.py
+      - name: Note version
+        shell: msys2 {0}
+        run: |
+          echo "PACKAGE_VERSION=$(python3 -- ./setup.py --version)" >> $GITHUB_ENV
+      - name: Install and package
+        shell: msys2 {0}
+        run: |
+          cd installer
+
+          # Download armips and other binary depedencies
+          ./download-windows-dependencies.sh
+
+          # Download and prepare the Arc theme
+          curl https://skytemple.org/build_deps/Arc.zip -O
+          unzip Arc.zip
+
+          # Install NSIS
+          curl https://skytemple.org/build_deps/nsis.zip -O
+          unzip -o nsis.zip -d "/c/Program Files (x86)/NSIS"
+
+          # Install PyInstaller
+          pip3 install pyinstaller
+
+          # Package
+          ./build-windows.sh $PACKAGE_VERSION
+
+      - name: Create installer
+        uses: joncloud/makensis-action@v3.4
+        with:
+          script-file: installer/skytemple.nsi
+          arguments: "/DPRODUCT_VERSION=${{ env.PACKAGE_VERSION }}"
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: skytemple-windows-app
+          path: |
+            installer/skytemple-*-install.exe
+
+  package-mac:
+    needs: build
+    runs-on: macos-10.15
+    name: Build and package for Mac OS
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Rewrite version for dev if not tag
+        if: "!startsWith(github.ref, 'refs/tags/')"
+        run: |
+          perl -i -pe "s/__version__\s*=\s*'(.*?)(rc.*|a.*|post.*)?'/__version__='\1.dev${GITHUB_SHA::8}'/" setup.py
+      - name: Note version
+        run: |
+          echo "PACKAGE_VERSION=$(python3 -- ./setup.py --version)" >> $GITHUB_ENV
+      - name: Install and package
+        run: |
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" # install Homebrew
```

Comment:
Homebrew is apparently already installed, so we can get rid of this. https://github.com/actions/virtual-environments/blob/macOS-10.15/20201130.3/images/macos/macos-10.15-Readme.md
```suggestion
```